simulatingrisk/hawkdovemulti/model.py

```diff
@@ -194,6 +212,52 @@ def __init__(
         # store whether to compare cumulative payoff or since last adjustment round
         self.adjust_payoff = adjust_payoff
 
+    def _risk_level_in_bounds(self, value):
+        # check if a generated risk level is within bounds
+        return self.min_risk_level <= value <= self.max_risk_level
+
+    def get_risk_attitude_generator(self):
+        """return a generator that will return risk attitudes for individual
+        agents based on the configured distribution."""
+        if self.risk_distribution == "uniform":
+            # uniform/random: generate random integer within risk level range
+            while True:
+                yield self.random.randint(self.min_risk_level, self.max_risk_level)
+        if self.risk_distribution == "normal":
+            # return values from a normal distribution centered around 4
+            while True:
+                yield int(self.random.gauss(4, 1.5))
+        elif self.risk_distribution == "skewed left":
+            # return values from a triangler distribution centered around 0
+            while True:
+                yield round(
+                    self.random.triangular(self.min_risk_level, self.max_risk_level, 0)
+                )
+        elif self.risk_distribution == "skewed right":
+            # return values from a triangler distribution centered around 8
+            while True:
+                yield round(
+                    self.random.triangular(self.min_risk_level, self.max_risk_level, 8)
+                )
+        elif self.risk_distribution == "bimodal":
+            # to generate a bimodal distribution, alternately generate
+            # values from two different normal distributions centered
+            # around the beginning and end of our risk attitude range
+            while True:
+                yield int(self.random.gauss(1, 0.75))
+                yield int(self.random.gauss(7, 0.75))
```

Comment:
what a cool idea for a bimodal distribution using generators. "bimodal" can then mean programmatically just alternating between the two modes ! very cool.

Reply:
This is a cool idea! Nice work.
Because r now ranges between 0 and 9, either the distributions should be centered around 1 and 8 or around 2 and 7 (I think the former is better because it leads to more extreme values--as long as it's still possible to get all the values).